docs/source/library-user-guide/api-health.md

```diff
@@ -69,3 +69,32 @@ For example:
 Deprecated methods will remain in the codebase for a period of 6 major versions or 6 months, whichever is longer, to provide users ample time to transition away from them.
 
 Please refer to [DataFusion releases](https://crates.io/crates/datafusion/versions) to plan ahead API migration
+
+## Migration Guidelines
+
+To ensure smooth upgrades and maintain application stability, the following guidelines should be followed for changes involving:
+
+- Public API changes
+- Introducing deprecated methods
+- Removal of obsolete methods
+
+Highlight all breaking changes, deprecated methods, and obsolete methods in the [migration guide](../../../MIGRATION_GUIDE.md).
+
+### Migration Document Requirements:
+
+For each upgrade, append a section in [migration guide](../../../MIGRATION_GUIDE.md) outlining the following:
```

Comment:
will we have merge conflicts there often?

Reply:
of course, like any other file, but should we easily resolvable.